### ISSUE TITLE
why p_mean_variance use noise_level instead of sample_gammas like in training for time conditon of denoise function.

### ISSUE THREAD

**Issue by zwmCool** [Janspiry/Palette-Image-to-Image-Diffusion-Models#89]
training:

`    
 def forward(self, y_0,group,y_cond=None, mask=None, noise=None):

        # sampling from p(gammas)

        b, *_ = y_0.shape
        t = torch.randint(1, self.num_timesteps, (b,), device=y_0.device).long()
        
        gamma_t1 = extract(self.gammas, t-1, x_shape=(1, 1))
        sqrt_gamma_t2 = extract(self.gammas, t, x_shape=(1, 1))
        sample_gammas = (sqrt_gamma_t2-gamma_t1) * torch.rand((b, 1), device=y_0.device) + gamma_t1
        sample_gammas = sample_gammas.view(b, -1)
        ....
        noise_hat = self.denoise_fn(torch.cat([y_cond, y_noisy*mask+(1.-mask)*y_0], dim=1), sample_gammas)
`

inference:

`

 def p_mean_variance(self, y_t, t,group, clip_denoised: bool, y_cond=None)

        noise_level = extract(self.gammas, t, x_shape=(1, 1)).to(y_t.device)

        y_0_hat = self.predict_start_from_noise(y_t, t=t, noise=self.denoise_fn(torch.cat([y_cond, y_t], dim=1), noise_level))
`

**Comment by minghuix98** [comment]
I have the same question. Did you find the reason?

**Comment by kirin-from-16** [comment]
#50 i think the author explain this in here.

**Comment by minghuix98** [comment]
> #50 i think the author explain this in here.

Got it. Thanks~
